pennylane/capture/primitives.py

```diff
@@ -117,3 +199,79 @@ def _(*_, **__):
         return abstract_type()
 
     return primitive
+
+
+def create_measurement_obs_primitive(
+    measurement_type: Type["qml.measurements.MeasurementProcess"], name: str
+) -> Optional["jax.core.Primitive"]:
+    """Create a primitive corresponding to the input type where the abstract inputs are an operator."""
+    if not has_jax:
+        return None
+
+    primitive = jax.core.Primitive(name + "_obs")
+
+    @primitive.def_impl
+    def _(obs, **kwargs):
+        return type.__call__(measurement_type, obs=obs, **kwargs)
+
+    abstract_type = _get_abstract_measurement()
+
+    @primitive.def_abstract_eval
+    def _(*_, **__):
+        abstract_eval = measurement_type._abstract_eval  # pylint: disable=protected-access
+        return abstract_type(abstract_eval, n_wires=None)
```

Comment:
Why are we not inferring `n_wires` from `obs` here? :thinking:

Reply:
Because we can't.  Suppose we have `qml.X(w1) + qml.Y(w2)`.  How many wires is that on? We would need concrete wires to be able to make that determination.  

So we also use `n_wires=None` to communicate to `_abstract_eval` that the measurement is defined via observable.

Reply:
Potentially we could just add a `has_obs` keyword argument to keep track of instead of using `n_wires=None`, it that would be more intuitive.

Reply:
Thanks for the explanation, makes sense! I recall this discussion now :D

Maybe `has_obs` would be nice, because we get the chance to have a clear internal API here, so we should use it if it makes sense :thinking: I actually don't have any concrete preferences, I'd just like to have specifiers for the different scenarios that are independent from the priorities of the different inputs with which they define the scenarios. Concretely, `n_wires=None` seems like information about `wires` but (also) is information about the `obs` in this case, and this logic mostly makes sense if `wires` take precedence over `obs`, I suppose?